chains/evm/contracts/test/feeQuoter/FeeQuoter.parseSuiExtraArgsFromBytes.t.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {FeeQuoter} from "../../FeeQuoter.sol";
+import {Client} from "../../libraries/Client.sol";
+
+import {Internal} from "../../libraries/Internal.sol";
+import {FeeQuoterSetup} from "./FeeQuoterSetup.t.sol";
+
+contract FeeQuoter_parseSuiExtraArgsFromBytes is FeeQuoterSetup {
+  FeeQuoter.DestChainConfig private s_destChainConfig;
+
+  function setUp() public virtual override {
+    super.setUp();
+    s_destChainConfig = _generateFeeQuoterDestChainConfigArgs()[0].destChainConfig;
+    s_destChainConfig.enforceOutOfOrder = true; // Enforcing out of order execution for messages to SVM
+    s_destChainConfig.chainFamilySelector = Internal.CHAIN_FAMILY_SELECTOR_SUI;
+
+    FeeQuoter.DestChainConfigArgs[] memory destChainConfigs = new FeeQuoter.DestChainConfigArgs[](1);
+    destChainConfigs[0] =
+      FeeQuoter.DestChainConfigArgs({destChainSelector: DEST_CHAIN_SELECTOR, destChainConfig: s_destChainConfig});
+    s_feeQuoter.applyDestChainConfigUpdates(destChainConfigs);
+  }
+
+  function test_SuiExtraArgsV1TagSelector() public pure {
+    assertEq(Client.SUI_EXTRA_ARGS_V1_TAG, bytes4(keccak256("CCIP SuiExtraArgsV1")));
+  }
+
+  function test_SuiExtraArgsV1() public view {
+    Client.SuiExtraArgsV1 memory inputArgs = Client.SuiExtraArgsV1({
+      gasLimit: GAS_LIMIT,
+      allowOutOfOrderExecution: true,
+      tokenReceiver: bytes32(uint256(0)),
+      receiverObjectIds: new bytes32[](2)
+    });
+
+    bytes memory inputExtraArgs = Client._suiArgsToBytes(inputArgs);
+
+    Client.SuiExtraArgsV1 memory expectedOutputArgs = Client.SuiExtraArgsV1({
```

Comment:
This is exactly the input struct, you don't have to declare it twice, you can just reference the first one

Reply:
resolved https://github.com/smartcontractkit/chainlink-ccip/pull/1153/commits/86aecbebb753c5624c450bfdc24831d098dd7a85